### ISSUE TITLE
Refactor dashboard cell context menu UX into on-hover icons

### ISSUE THREAD

**Issue by jaredscheib** [influxdata/chronograf#1964]
It feels somewhat unintuitive as to how a user would discover the ability to Edit a dashboard cell. In discussion with @alexpaxton, we thought it would be clearer UX to understand how to Edit a cell and perform other context operations by adding icons to the top-right of a cell, on-hover, such as a Pencil for Edit, etc.

Also, this probably should be tackled after #1965 so double work isn't done.

**Comment by alexpaxton** [comment]
Should moving rename into cell edit be a separate issue? As mentioned, this issue is contingent upon that change

**Comment by jaredscheib** [comment]
Done -- thanks
